cpu/esp32/esp-now/esp_now_netif.c

```diff
@@ -180,6 +180,6 @@ void auto_init_esp_now(void)
             gnrc_netif_esp_now_create(_esp_now_stack,
                                       ESP_NOW_STACKSIZE, ESP_NOW_PRIO,
                                       "net-esp-now",
-                                      (netdev_t *)&esp_now_dev);
+                                      &esp_now_dev->netdev);
```

Comment:
I saw that `_gnrc_netif_thread` tried to call a NULL pointer here
https://github.com/RIOT-OS/RIOT/blob/7a5142f7bcaf2ec35de74774ea5b1beaeae64caa/sys/net/gnrc/netif/gnrc_netif.c#L1259

Reply:
I'm kinda surprised it didn't crash. Or, well, I guess it did, but without printing a single message.
What made debugging this annoying was that printf() does not flush on newlines it seems, so I'd always had to print enough to flush the buffer being able to see what's actually going on. fflush(stdout) does nothing for me.

Reply:
Yes, `printf` debugging is sometimes a bit difficult. The reason is that the UART transmission is executed asynchronously in hardware. Since the transmission buffer has a size of 128 characters, most `printf` outputs can be written to the transmission buffer by a thread without any waiting. Thus, the thread can continue while the UART transmission is still going on.
The only way to see at which line of code the execution hangs is to add a short sleep with `xtimer_usleep` after the `printf` so that all characters can be sent before the thread continues.